src/Aspire.Hosting.Azure.Sql/AzureSqlExtensions.cs

```diff
@@ -127,6 +129,18 @@ public static IResourceBuilder<AzureSqlDatabaseResource> AddDatabase(this IResou
         }
     }
 
+    /// <summary>
+    /// Configures the Azure SQL Database to be deployed with the specified SKU
+    /// </summary>
+    /// <param name="builder">The builder for the Azure SQL resource.</param>
+    /// <param name="skuName">SKU of the database. If not provided, this defaults to the free database tier.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<AzureSqlDatabaseResource> WithSku(this IResourceBuilder<AzureSqlDatabaseResource> builder, string skuName)
```

Comment:
Do we need both the optional parameter overload above and a `WithSku` method? Can we just have a new `WithSku` method instead?